### PR TITLE
master wowl report ges

### PR DESCRIPTION
**Pull request by SimonGenin**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/account/views/report_invoice.xml [hunk 1]
```diff
@@ -73,7 +73,7 @@
 
                                 <tr t-att-class="'bg-200 font-weight-bold o_line_section' if line.display_type == 'line_section' else 'font-italic o_line_note' if line.display_type == 'line_note' else ''">
                                     <t t-if="not line.display_type" name="account_invoice_line_accountable">
-                                        <td name="account_invoice_line_name"><span t-field="line.name" t-options="{'widget': 'text'}"/></td>
+                                        <td t-att-res-id="line.id" t-att-view-type="'form'" t-att-res-model="'account.move.line'"  name="account_invoice_line_name"><span t-field="line.name" t-options="{'widget': 'text'}"/></td>
```
Comment (SimonGenin) [inline addons/account/views/report_invoice.xml]: Just a change that allows trying the actions from client actions.

Comment (kebeclibre) [reply addons/account/views/report_invoice.xml]: fair enough, though the feature should be tested in its own right ; leaving this file alone for merging in master

File: addons/web/static/src/report/report.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+            const action = {
+                type: "ir.actions.act_window",
+                view_mode: element.getAttribute("view-type"),
+                res_id: Number(element.getAttribute("res-id")),
+                res_model: element.getAttribute("res-model"),
+                views: [[element.getAttribute("view-id"), element.getAttribute("view-type")]],
+            };
+            window.parent.postMessage(
+                {
+                    message: "report:do_action",
```
Comment (kebeclibre) [inline addons/web/static/src/report/report.js]: report:doAction

File: addons/web/static/src/report/report_client_action.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        this.title = this.props.options.display_name || this.props.options.name;
+        this.trusted_origin = reportUtils.buildOrigin(
+            reportUtils.getProtocolFromUrl(this.userService.web_base_url),
+            reportUtils.getHostFromUrl(this.userService.web_base_url)
+        );
+        this.report_url = this.props.options.report_url;
+    }
+
+    mounted() {
+        window.addEventListener("message", this.onMessageReceived.bind(this));
```
Comment (kebeclibre) [inline addons/web/static/src/report/report_client_action.js]: useExternalListener in constructor ?
And should not we bind the handler on the iframe ? by using HTMLIFrameElement.contentWindow (to reference an embedded <iframe> from its parent window), from the clientAction and not the other way around ?

File: addons/web/static/src/report/utils.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+/** @odoo-module **/
+
+export function getProtocolFromUrl(url) {
```
Comment (kebeclibre) [inline addons/web/static/src/report/utils.js]: use the URL native API ?https://developer.mozilla.org/en-US/docs/Web/API/URL

File: addons/web/__manifest__.py [hunk 5]
```diff
@@ -233,29 +236,34 @@
         'web.report_assets_common': [
             ('include', 'web._assets_helpers'),
 
-            'web/static/src/legacy/scss/bootstrap_overridden_report.scss',
+            'web/static/src/report/styles/bootstrap_overridden_report.scss',
```
Comment (ged-odoo) [inline addons/web/__manifest__.py:234]: why not `'web/static/src/report/bootstrap_overridden_report.scss'`? I think there is no real need to scope the style files in report/

**Review by kebeclibre — CHANGES_REQUESTED** [review]

File: addons/web/__manifest__.py [hunk 6]
```diff
@@ -233,29 +236,34 @@
...
+
             'web/static/src/legacy/js/services/session.js',
             'web/static/src/legacy/js/public/public_root.js',
             'web/static/src/legacy/js/public/public_root_instance.js',
             'web/static/src/legacy/js/public/public_widget.js',
-            'web/static/src/legacy/js/report/utils.js',
-            'web/static/src/legacy/js/report/report.js',
+
+            'web/static/src/report/utils.js',
+            'web/static/src/report/report.js',
```
Comment (ged-odoo) [inline addons/web/__manifest__.py]: it looks like it would be better if we could do a glob expression for report/

File: addons/web/static/src/core/user_service.js [hunk 7]
```diff
@@ -75,6 +75,7 @@ export const userService = {
             home_action_id,
             show_effect: showEffect,
         } = info;
+        const web_base_url = odoo.session_info['web.base.url'];
```
Comment (ged-odoo) [inline addons/web/static/src/core/user_service.js]: not sure about this

File: addons/web/__manifest__.py (hunk 5 above)
Comment (SimonGenin) [reply addons/web/__manifest__.py:234]: it's because there are like 10 scss files, that felt crowded

File: addons/web/static/src/report/report.js [hunk 8]
```diff
@@ -0,0 +1,47 @@
+/** @odoo-module **/
+
```
Comment (ged-odoo) [inline addons/web/static/src/report/report.js]: add a comment to make it clear that it is intended to be run in the report iframe

File: addons/web/static/src/report/report.js [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+import * as reportUtils from "./utils";
+const { utils } = owl;
+
+(async function () {
```
Comment (ged-odoo) [inline addons/web/static/src/report/report.js]: all of this should be a service

File: addons/web/static/src/js/report/client_action.js [hunk 10]
```diff
@@ -42,14 +42,14 @@ var ReportAction = AbstractAction.extend({
         this.controlPanelProps.cp_content = {
             $buttons: this.$buttons,
         };
-        return Promise.all([this._super.apply(this, arguments), session.is_bound]).then(async function () {
+        return Promise.all([this._super.apply(this, arguments), session.is_bound]).then(async () => {
```
Comment (ged-odoo) [inline addons/web/static/src/js/report/client_action.js]: why is this file moved into static/src/js/report?

File: addons/web/static/src/report/report_client_action.js [hunk 11]
```diff
@@ -0,0 +1,88 @@
+/** @odoo-module alias=report.client_action **/
+
+import ControlPanel from "web.ControlPanel";
+
+import { registry } from "@web/core/registry";
+import { breadcrumbsToLegacy } from "@web/legacy/utils";
```
Comment (ged-odoo) [inline addons/web/static/src/report/report_client_action.js]: non legacy code should not import legacy code... NO!!!

File: addons/web/static/src/report/report_client_action.js [hunk 12]
```diff
@@ -0,0 +1,88 @@
...
+            data: this.props.options.data || {},
+            context: this.props.options.context || {},
+            display_name: this.title,
+        });
+    }
+}
+ReportAction.components = { ControlPanel };
+// GES: FIXME Use @DAM Control Panel & remove the event handler
+ReportAction.template = tags.xml`
+    <div class="h-100">
```
Comment (ged-odoo) [inline addons/web/static/src/report/report_client_action.js]: should be in a xml file, so it can be translated

File: addons/web/static/src/report/styles/super.reset.min.scss [hunk 13]
```diff
@@ -0,0 +1,192 @@
+/*
```
Comment (ged-odoo) [inline addons/web/static/src/report/styles/super.reset.min.scss]: super.reset.min.scss?

**Comment by ged-odoo** [comment]
oh also, don't forget: i think there are pdf and other image files somewhere in static/ that should be moved into report/

File: addons/web/static/src/core/user_service.js (hunk 7 above)
Comment (SimonGenin) [reply addons/web/static/src/core/user_service.js]: Where would you put it ? In another service ? It seemed like most of the session_info stuff was put inside the user service so I did the same but I agree the web_base_url doesn't really belong there. 
Or maybe should I call `odoo.session_info['web.base.url']`straight from the component, bypassing everything ?

File: addons/web/static/src/js/report/client_action.js (hunk 10 above)
Comment (SimonGenin) [reply addons/web/static/src/js/report/client_action.js]: Because it is a client action specifically for reports, so it goes inside the report dir

File: addons/web/static/src/report/styles/super.reset.min.scss (hunk 13 above)
Comment (SimonGenin) [reply addons/web/static/src/report/styles/super.reset.min.scss]: oups

File: addons/web/static/tests/report/report_action_tests.js [hunk 14]
```diff
@@ -144,26 +148,29 @@ QUnit.module("ActionManager", (hooks) => {
...
             // it doesn't trigger an RPC when it is appended to the DOM (for this
             // usecase, using removeSRCAttribute doesn't work as the RPC is
             // triggered as soon as the iframe is in the DOM, even if its src
             // attribute is removed right after)
-            testUtils.mock.patch(ReportClientAction, {
-                async start() {
-                    await this._super(...arguments);
-                    this._rpc({ route: this.iframe.getAttribute("src") });
-                    this.iframe.setAttribute("src", "about:blank");
+            patch(ReportClientAction.prototype, "BlankReportClientActionIFrame", {
```
Comment (ged-odoo) [inline addons/web/static/tests/report/report_action_tests.js]: use `patchWithCleanup`

File: addons/web/static/tests/report/report_action_tests.js [hunk 15]
```diff
@@ -206,13 +213,14 @@ QUnit.module("ActionManager", (hooks) => {
         // usecase, using removeSRCAttribute doesn't work as the RPC is
         // triggered as soon as the iframe is in the DOM, even if its src
         // attribute is removed right after)
-        testUtils.mock.patch(ReportClientAction, {
```
Comment (ged-odoo) [inline addons/web/static/tests/report/report_action_tests.js]: same

**Comment by SimonGenin** [comment]
working on https://github.com/odoo-dev/odoo/pull/896
